### PR TITLE
Transplant refactored shared mode

### PR DESCRIPTION
**Pull request by lincsoon**
This patch transplants refactored shared mode from the master branch
to the lite branch and integrates decompressor into it.

### CODE REVIEW

**Comment by lincsoon** [comment]
@halton @d0ngjun @shawngao5

**Comment by lincsoon** [comment]
The decompressing part focuses on `XWalkActivity`, `XWalkViewDelegate`, `Decompressor`

**Comment by halton** [comment]
I'll reply on @d0ngjun @shawngao5 review.

File: build/android/generate_xwalk_core_library.py [hunk 1]
```diff
@@ -143,28 +151,13 @@ def CopyBinaries(out_dir, no_icu_data, use_lzma):
...
-    shutil.copyfile(source_file, target_file)
-
   # Copy native libraries.
   source_dir = os.path.join(out_dir, XWALK_CORE_SHELL_APK, 'libs')
   distutils.dir_util.copy_tree(source_dir, libs_dir)
 
   # NOTE: Gradle doesn't accept '-', use '_' instead.
   if use_lzma:
-    for arch in ['x86', 'armeabi_v7a']:
+    for arch in ['x86', 'armeabi-v7a']:
```
Comment (d0ngjun) [inline build/android/generate_xwalk_core_library.py]: Gradle will report error if we pack the library use armeabi-v7a suffix, please use armeabi_v7a instead.

Comment (lincsoon) [inline build/android/generate_xwalk_core_library.py]: Done

**Comment by lincsoon** [comment]
@halton @d0ngjun @shawngao5 Updated with new init interface. Please review

**Comment by d0ngjun** [comment]
LGTM

**Pull request merged**